### ISSUE TITLE
Using with on: [deployment_status]

### ISSUE THREAD

**Issue by gr2m** [cypress-io/github-action#150]
Our app is continuously deployed using [Vercel](https://vercel.com/). Every commit of every pull request is being deployed, so I want our Cypress tests to run against this deployment. For that reason, our test workflow is not triggered by `push` or `pull_request`, but by `deployment_status`. But we still want to create a commit status on the pull request at then end. [Our workflow file](https://github.com/gr2m/octokit.rest/blob/master/.github/workflows/test.yml) is currently looking like this:

```yml
name: End-to-end tests
on: [deployment_status]

jobs:
  cypress-run:
    runs-on: ubuntu-latest
    if: ${{ github.event.deployment_status.state == 'success' }}
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: debug
        uses: hmarr/debug-action@master
        env:
          TEST_URL: ${{ github.event.deployment_status.target_url }}
      - name: Cypress run
        id: cypress
        uses: cypress-io/github-action@v1
        # Continue the build in case of an error, as we need to set the
        # commit status in the next step, both in case of success and failure
        continue-on-error: true
        with:
          spec: cypress/integration/happy.spec.js
          record: true
          env: base_url=${{ github.event.deployment_status.target_url }}
          ci-build-id: ${{ github.sha }}
        env:
          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      # Action does not set commit/pull reques status automatically, because it
      # is not triggered by the `push` event. So we have to set the commit
      # status manually.
      - name: Create commit status
        uses: octokit/request-action@v2.x
        id: create_status
        with:
          route: POST /repos/:repository/statuses/:sha
          repository: ${{ github.repository }}
          sha: ${{ github.sha }}
          context: test
          description: Cypress run
          state: ${{ steps.cypress.outcome}}
          target_url: https://dashboard.cypress.io/projects/4nc4rv/runs
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

And setting the status works as expected:

<img width="959" alt="image" src="https://user-images.githubusercontent.com/39992/83331919-6b9f3980-a24d-11ea-9082-e21f3df75dd8.png">

But I would like to set the `target_url` to the actual run URL, e.g. https://dashboard.cypress.io/projects/4nc4rv/runs/11/specs. But I cannot find a way to know that URL, because I don't know the run number `11` in the `create_status` step. 

GitHub Actions support setting outputs for that purpose:
https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter. If `cypress-io/github-action` knows the run URL, could you set it to an output parameter?

---

Related question: `Branch` is currently `undefined` in our runs: https://dashboard.cypress.io/projects/4nc4rv/runs/11/specs, I guess the reason is that the build is not triggered by `push` or `pull_request`. Is there a way to set the branch name manually from my side?

**Comment by bahmutov** [comment]
Great issue I will look into this next week

**Comment by bahmutov** [comment]
@gr2m for the `Branch` you can pass it "manually" by setting an environment variable https://docs.cypress.io/guides/guides/continuous-integration.html#Git-information

```
env:
  COMMIT_INFO_BRANCH: your name here
```

**Comment by bahmutov** [comment]
Dashboard URL is blocked by https://github.com/cypress-io/cypress/issues/7532

**Comment by bahmutov** [comment]
I have released v2 of this action https://github.com/cypress-io/github-action/releases/tag/v2.0.0 that now runs Cypress via its NPM module API. This will allow us to export the dashboard URL at the end of the run, so that's the next feature to implement :)

**Comment by bahmutov** [comment]
@gr2m Gregor, I think instead of `continue_on_error: true` which 'loses' the test status, why not set the next command to always run using `if` notation like in https://github.com/cypress-io/github-action#artifacts

```yml
- uses: cypress-io/github-action@v2
# Test run video was always captured, so this action uses "always()" condition
- uses: actions/upload-artifact@v1
  if: always()
  with:
    name: cypress-videos
    path: cypress/videos
```

**Comment by gr2m** [comment]
> which 'loses' the test status

Can you clarify what you mean? `steps.cypress.outcome` is set correctly for me when using `continue-on-error: true`

**Comment by bahmutov** [comment]
I mean - yes, it is passed correctly. But I think allowing the step to fail, and yet running a step after that is better than `continue_on_error`.

**Comment by gr2m** [comment]
Got it now, thanks :)

**Comment by chinchang** [comment]
The following approach is lot better -> https://twitter.com/mikenikles/status/1272138850333327360?s=20
Works with `push/pull` event

**Comment by verheyenkoen** [comment]
> Related question: `Branch` is currently `undefined` in our runs: https://dashboard.cypress.io/projects/4nc4rv/runs/11/specs, I guess the reason is that the build is not triggered by `push` or `pull_request`. Is there a way to set the branch name manually from my side?

The [docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#deployment_status) seem to state that the GITHUB_REF would indeed be empty if triggered with a commit ID. However there should be a way to retrieve the branch name from a commit ID. If you open any commit on GitHub, you can see the corresponding commit in the header. Would also like a solution for that. In our case we also see "Unknown branch" in the Cypress Dashboard.

> @gr2m for the `Branch` you can pass it "manually" by setting an environment variable https://docs.cypress.io/guides/guides/continuous-integration.html#Git-information
> 
> ```
> env:
>   COMMIT_INFO_BRANCH: your name here
> ```

Hard coding it in the GitHub Action yaml file isn't really a solution I guess. You want this to reflect the actual branch.

**Comment by MikeMcC399** [comment]
@verheyenkoen 

If you are expecting something to be done based on your comments then I would suggest that you open a new issue as this one is closed with an enhancement that was already implemented as a feature in `v2` and the current version of the action is now `v5`.

Were you intending to submit a feature request for this use-case?

**Comment by verheyenkoen** [comment]
@MikeMcC399 Not sure. I guess this is more a GitHub Actions issue than a Cypress issue. But maybe Cypress can improve this anyway. Was looking for a way to retrieve the branch name for a "deployment_status" event trigger and stumbled upon this ticket.

I guess the flow is, a GitHub push triggers a vercel build and that one triggers the workflow run, so I understand there is no real commit context for GitHub Action to work in but, if it has already the commit ID/hash, it should be possible to retrieve the branch name from that, no?

**Comment by bahmutov** [comment]
This is not something can grab on its own I think, instead it is something Vercel should provide with the event, if GitHub allows Sent from my iPhoneOn May 10, 2023, at 08:57, Koen Verheyen ***@***.***> wrote:﻿
@MikeMcC399 Not sure. I guess this is more a GitHub Actions issue than a Cypress issue. But maybe Cypress can improve this anyway. Was looking for a way to retrieve the branch name for a "deployment_status" event trigger and stumbled upon this ticket.
I guess the flow is, a GitHub push triggers a vercel build and that one triggers the workflow run, so I understand there is no real commit context for GitHub Action to work in but, if it has already the commit ID/hash, it should be possible to retrieve the branch name from that, no?
